File: Relatório 2/ex2.lua
print("Quantos valores deseja inserir?")
local N = tonumber(io.read())
local tabela = {}

for i = 1, N do
    print("Digite o valor " .. i .. ":")
    tabela[i] = tonumber(io.read())
end

local function buscarMaior(t)
    local maior = t[1]
    for i = 2, #t do
        if t[i] > maior then
            maior = t[i]
        end
    end
    return maior
end

print("O maior valor encontrado é: " .. buscarMaior(tabela))

File: Relatório 05/ex2.cpp
#include <iostream>
using namespace std;

float multiplicaArray(float arr[], int tamanho) {
    float produto = 1.0f;
    for (int i = 0; i < tamanho; i++) {
        produto *= arr[i];
    }
    return produto;
}

int main() {
    int n;
    cout << "Quantos valores deseja inserir? ";
    cin >> n;
    
    float* values = new float[n];
    for (int i = 0; i < n; i++) {
        cout << "Valor " << i + 1 << ": ";
        cin >> values[i];
    }
    
    cout << "Produto: " << multiplicaArray(values, n) << endl;
    
    delete[] values;
    return 0;
}
